Sources/Baggage/Baggage.swift

```diff
@@ -0,0 +1,209 @@
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+// ==== ----------------------------------------------------------------------------------------------------------------
+// MARK: Baggage
+
+/// A `Baggage` is a heterogeneous storage type with value semantics for keyed values in a type-safe fashion.
+///
+/// Its values are uniquely identified via `BaggageKey`s (by type identity). These keys also dictate the type of
+/// value allowed for a specific key-value pair through their associated type `Value`.
+///
+/// ## Defining keys and accessing values
+/// Baggage keys are defined as types, most commonly case-less enums (as no actual instances are actually required)
+/// which conform to the `Baggage.Key` protocol:
+///
+///     private enum TestIDKey: BaggageKey {
+///       typealias Value = String
+///     }
+///
+/// While defining a key, one should also immediately declare an extension on `Baggage`,
+/// to allow convenient and discoverable ways to interact with the baggage item, the extension should take the form of:
+///
+///     extension Baggage {
+///       var testID: TestIDKey.Value? {
+///         get {
+///           self[TestIDKey.self]
+///         } set {
+///           self[TestIDKey.self] = newValue
+///         }
+///       }
+///     }
+///
+/// For consistency, it is recommended to name key types with the `...Key` suffix (e.g. `SomethingKey`) and the property
+/// used to access a value identifier by such key the prefix of the key (e.g. `something`). Please also observe the usual
+/// Swift naming conventions, e.g. prefer `ID` to `Id` etc.
+///
+/// ## Usage
+/// Using a baggage container is fairly straight forward, as it boils down to using the prepared computed properties:
+///
+///     var context = Baggage.background
+///     // set a new value
+///     context.testID = "abc"
+///     // retrieve a stored value
+///     let testID = context.testID ?? "default"
+///     // remove a stored value
+///     context[TestIDKey.self] = nil
+///
+/// Note that normally a baggage should not be "created" ad-hoc by user code, but rather it should be passed to it from
+/// a runtime. For example, when working in an HTTP server framework, it is most likely that the baggage is already passed
+/// directly or indirectly (e.g. in a `BaggageContext` or `FrameworkContext`)
+///
+/// ### Accessing all values
+///
+/// The only way to access "all" values in a baggage context is by using the `forEachBaggageItem` function.
+/// The baggage container on purpose does not expose more functions to prevent abuse and treating it as too much of an
+/// arbitrary value smuggling container, but only make it convenient for tracing and instrumentation systems which need
+/// to access either specific or all items carried inside a baggage.
+public struct Baggage {
```

Comment:
okey then, here we go: as simple as it can possibly be.
there's no way to make "your context" get `.traceID` now, people will have to `yourContext.baggage.traceID` but I guess that is fine TBH since their main purpose is to "set it once, and forget it exists".

 @tanner0101 @tomerd

Reply:
I am +1 on this simple, explicit form